server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -394,6 +412,13 @@ private static ClusterState createFailureStoreIndex(
             return currentState;
         }
 
+        var settingsBuilder = Settings.builder().put(IndexMetadata.SETTING_INDEX_HIDDEN, true);
```

Comment:
Maybe only call the builder if the `refreshInterval` is not null. This way we can avoid creating new objects when they are not necessary.